src/main/scala/io/iohk/ethereum/testmode/TestEthBlockServiceWrapper.scala

```diff
@@ -32,9 +33,19 @@ class TestEthBlockServiceWrapper(
     .getByBlockHash(request)
     .map(
       _.map(blockByBlockResponse => {
-        val fullBlock = blockchainReader.getBlockByNumber(blockByBlockResponse.blockResponse.get.number).get
-        BlockByBlockHashResponse(blockByBlockResponse.blockResponse.map(response => toEthResponse(fullBlock, response)))
-      })
+        import io.iohk.ethereum.utils.ByteStringUtils._
+        blockByBlockResponse.blockResponse
+          .toRight(s"EthBlockService: unable to find block for hash ${request.blockHash.toHex}")
+          .flatMap(baseBlockResponse => baseBlockResponse.hash.toRight(s"missing hash for block $baseBlockResponse"))
+          .flatMap(hash => ledger.getBlockByHash(hash).toRight(s"Ledger: unable to find block for hash=${hash.toHex}"))
+          .map(fullBlock =>
+            BlockByBlockHashResponse(
+              blockByBlockResponse.blockResponse.map(response => toEthResponse(fullBlock, response))
+            )
+          )
+          .left
+          .map(errorMessage => JsonRpcError.LogicError(errorMessage))
+      }).flatten
```

Comment:
You can make a few simplifications here:
- move the import to the top of the file, please
- remove the final `.flatten` and use `flatMap` instead
- you can remove `{` after `=>`, by replacing the `(` before `blockByBlockResponse` by `{`
In summary from:
`_.map(blockByBlockResponse => {`  to `_.flatMap { blockByBlockResponse =>`

Reply:
thanks, update done